src/api/tasks/tasksApi.ts

```diff
@@ -0,0 +1,129 @@
+import "@pnp/sp/webs";
+import "@pnp/sp/lists";
+import "@pnp/sp/batching";
+import { spWebContext, subWebContext } from "src/api/SPWebContext";
+import { useMutation, useQuery } from "@tanstack/react-query";
+import { z } from "zod";
+import { WorkflowDetails } from "src/components/ViewPCOL/Actions/StartForm/StartWorkflow";
+
+const Task = z.object({
+  Id: z.number().positive(),
+  Title: z.string(),
+  pcolId: z.string(),
+  Person: z.object({
+    Id: z.number().int().positive(),
+    Title: z.string(),
+    EMail: z.string(),
+  }),
+  Role: z.string(),
+  Status: z.string().optional(),
+  SkippedBy: z
+    .object({
+      Id: z.number().int().positive(),
+      Title: z.string(),
+      EMail: z.string(),
+    })
+    .optional(),
+  Modified: z.coerce.date().optional(),
+});
+type Task = z.infer<typeof Task>;
+
+export const useTasks = (subSite: string, pcolId: string) => {
+  return useQuery({
+    queryKey: ["tasks", subSite, pcolId],
+    queryFn: () =>
+      subWebContext(subSite)
+        .web.lists.getByTitle("tasks")
+        .items.select(
+          "Id",
+          "Title",
+          "pcolId",
+          "Person/Id",
+          "Person/Title",
+          "Person/EMail",
+          "Role",
+          "Status",
+          "SkippedBy/Id",
+          "SkippedBy/Title",
+          "SkippedBy/EMail",
+          "Modified"
+        )
+        .expand("Person", "SkippedBy")<Task[]>(),
+  });
+};
+
+export const useAddTasks = (subSite?: string, pcolId?: string) => {
+  return useMutation({
+    mutationFn: async (wfDetails: WorkflowDetails) => {
+      if (!subSite) {
+        return Promise.reject("Invalid site");
+      }
+      if (!pcolId) {
+        return Promise.reject("Invalid PCOL");
+      }
+      const [batched, execute] = subWebContext(subSite).batched();
+      const batch = batched.web.lists.getByTitle("tasks");
+
+      wfDetails.ParallelReviewers.forEach(async (reviewer) => {
```

Comment:
Not sure if there is a better way to structure these loops. The forEach doesn't wait for the await to finish before moving on, however it seems like maybe batch handles it okay, as it still queues the item, but since it is waiting for a call to finish, each of these Serial/Parallel are then fired off as separate network calls rather than actually being part of a bigger batch network call.